csharp/src/Ice/TlsServerOptions.cs

```diff
@@ -0,0 +1,93 @@
+        /// <see cref="ClientCertificateValidationCallback"/>.</summary>
+        public X509Certificate2Collection? ClientCertificateCertificateAuthorities
+        {
+            get => _clientCertificateCertificateAuthorities;
+            set
+            {
+                if (ClientCertificateValidationCallback != null)
+                {
+                    throw new ArgumentException(
+                        @$"Using the {nameof(ClientCertificateCertificateAuthorities)} is incompatible with using the {
+                           nameof(ClientCertificateValidationCallback)}");
+                }
+                _clientCertificateCertificateAuthorities = value;
+            }
+        }
+
+        /// <summary>Gets or sets the callback that will be used to verify the client certificate used for
+        /// authentication. Setting this is incompatible with setting
+        /// <see cref="ClientCertificateCertificateAuthorities"/>.</summary>
+        public RemoteCertificateValidationCallback? ClientCertificateValidationCallback
+        {
+            get => _clientCertificateValidationCallback;
+            set
+            {
+                if (CertificateRevocationCheckMode != null)
+                {
+                    throw new ArgumentException(
+                        @$"Using the {nameof(ClientCertificateValidationCallback)} is incompatible with using the {
+                           nameof(CertificateRevocationCheckMode)}");
+                }
+
+                if (ClientCertificateCertificateAuthorities != null)
+                {
+                    throw new ArgumentException(
+                        @$"Using the {nameof(ClientCertificateValidationCallback)} is incompatible with using the {
+                           nameof(ClientCertificateCertificateAuthorities)}");
+                }
+
+                _clientCertificateValidationCallback = value;
+            }
+        }
+
+        /// <summary>The list of SSL protocols to enable for incoming connections, the default is None that allows the
+        /// OS to choose the best protocol to use. It can be set to Tls12, Tls13, None or a combination of those, other
+        /// values are not accepted.</summary>
+        public SslProtocols? EnabledSslProtocols { get; set; }
+
+        /// <summary>Gets or sets a boolean value that specifies whether the client is asked for a certificate for
+        /// authentication.</summary>
+        public bool RequireClientCertificate { get; set; }
+
+        /// <summary>Gets or sets the certificate user for incoming connections.</summary>
+        public X509Certificate2? ServerCertificate { get; set; }
+
+        private X509RevocationMode? _certificateRevocationCheckMode;
+        private RemoteCertificateValidationCallback? _clientCertificateValidationCallback;
+        private X509Certificate2Collection? _clientCertificateCertificateAuthorities;
+    }
+}
```

Comment:
Ditto for validation.

Reply:
For validation if there is two incompatible properties the first you set is accepted and the second throws, what do you suggest, call a internal `validate` after anything each setter?

Reply:
Yes, an internal `validate`. Or just document that with the callback that other properties are ignored if this is set.

Reply:
I keeped the `ArgumentException` in the setters there is only two incompatible properties now